Lib/functools.py

```diff
@@ -273,43 +273,130 @@ def reduce(function, sequence, initial=_initial_missing):
 ### partial() argument application
 ################################################################################
 
+
+class __PlaceholderTypeBase:
```

Comment:
Isn't it overcomplication? Why not simply declare a class and create its single instance? We can add a simple guard to make it non-instantiable:
```py
@objec.__new__
class Placeholder:
    __new__ = None
```
We should not bother with exposing the type of `Placeholder` and making it pickleable.

Reply:
I think the latest implementation of this (thanks to @XuehaiPan) is pretty simple and is in a very good alignment with the one in `C`.

Reply:
I don't think pickle update complicates things much.

`functools` is already being imported to both Python and C pickle modules.

`PlaceholderType` serialization is a trivial addition, which ensures consistency with `None` type constants.